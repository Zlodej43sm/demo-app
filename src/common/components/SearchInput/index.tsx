import React, { ComponentType } from "react";
import { useIntl } from "react-intl";
import { Box, Input } from "@mui/material";
import SearchIcon from "@mui/icons-material/Search";

import useStyles from "./styles";

type Props = {
  onSearch: () => void;
};

const SearchInput: ComponentType<Props> = ({ onSearch }) => {
  const { formatMessage } = useIntl();
  const classes = useStyles();

  return (
    <Box className={classes.root}>
      <Input
        fullWidth
        disableUnderline
        onInput={onSearch}
        startAdornment={
          <SearchIcon className={classes.searchIconRoot} fontSize="small" />
        }
        classes={{ root: classes.inputWrapperRoot }}
        placeholder={formatMessage({ id: "input.filter.placeholder" })}
        type="search"
      />
    </Box>
  );
};

export default SearchInput;
