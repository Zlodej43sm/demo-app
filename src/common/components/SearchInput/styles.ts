import { makeStyles } from "@mui/styles";

export const COMPONENT_NAME = "SearchInput";

export default makeStyles(
  (theme: any) => ({
    root: {
      width: "100%",
      position: "relative",
      borderRadius: 5,
      minHeight: 29,
      backgroundColor: theme.palette.grey["100"],
      marginBottom: theme.spacing(2),
    },
    searchIconRoot: {
      position: "absolute",
      left: theme.spacing(),
      color: theme.palette.grey["300"],
    },
    inputWrapperRoot: {
      "& .MuiInputBase-input": {
        position: "relative",
        zIndex: 333,
        padding: theme.spacing(0.75, 0.75, 0.75, 4.5),
        fontSize: "12px",
      },
    },
  }),
  { name: COMPONENT_NAME }
);
