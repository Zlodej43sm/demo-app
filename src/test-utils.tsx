import React, { ComponentType, PropsWithChildren } from "react";
import { createIntl, RawIntlProvider } from "react-intl";
import { render, RenderOptions } from "@testing-library/react";
import { ThemeProvider, createTheme } from "@mui/material/styles";

import { LANG_CODES } from "./common/constants";
import { appMessages } from "./lang";

const theme = createTheme({
  palette: {},
});
const intl = createIntl({
  locale: LANG_CODES.EN,
  messages: appMessages(LANG_CODES.EN),
});

const AllTheProviders: ComponentType<PropsWithChildren> = ({
  children,
}): any => (
  <ThemeProvider theme={theme}>
    <RawIntlProvider value={intl}>{children}</RawIntlProvider>
  </ThemeProvider>
);

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";

const mockAppStore = {
  spreadsheet: {
    data: [
      [
        {
          id: "A0",
          value: "3",
          formula: "",
        },
        {
          id: "B0",
          value: "18",
          formula: "",
        },
        {
          id: "C0",
          value: 13.5,
          formula: "=12 + 3 / (2)",
        },
      ],
    ],
    byId: {
      A0: [0, 0],
      B0: [0, 1],
      C0: [0, 2],
    },
    progress: {},
    loading: false,
    error: false,
  },
};

export { customRender as render, mockAppStore };
