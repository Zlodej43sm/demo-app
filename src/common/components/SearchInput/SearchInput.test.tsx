import React from "react";

import { render } from "../../../test-utils";
import SearchInput from ".";

describe("SearchInput", () => {
  const props = {
    onSearch: jest.fn(),
  };

  it("SHOULD match snapshot", () => {
    const { container } = render(<SearchInput {...props} />);
    expect(container).toMatchSnapshot();
  });
});
