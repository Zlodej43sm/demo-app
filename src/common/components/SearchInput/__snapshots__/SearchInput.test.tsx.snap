// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput SHOULD match snapshot 1`] = `
<div>
  <div
    class="SearchInput-root-1 MuiBox-root css-0"
  >
    <div
      class="MuiInputBase-root MuiInput-root SearchInput-inputWrapperRoot-3 MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-adornedStart css-1s97tdc-MuiInputBase-root-MuiInput-root"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall SearchInput-searchIconRoot-2 css-ptiqhd-MuiSvgIcon-root"
        data-testid="SearchIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
      <input
        class="MuiInputBase-input MuiInput-input MuiInputBase-inputTypeSearch MuiInputBase-inputAdornedStart css-c63i49-MuiInputBase-input-MuiInput-input"
        placeholder="Type a search query to filter"
        type="search"
        value=""
      />
    </div>
  </div>
</div>
`;
